test/integration/webhook/core/workload_test.go

```diff
@@ -734,6 +749,50 @@ var _ = ginkgo.Describe("Workload validating webhook", func() {
 				},
 				gomega.Succeed(),
 			),
+			ginkgo.Entry("can add maximum execution time when not admitted",
+				func() *kueue.Workload {
+					return testing.MakeWorkload(workloadName, ns.Name).Obj()
+				},
+				false,
+				func(newWL *kueue.Workload) {
+					newWL.Spec.MaximumExecutionTimeSeconds = ptr.To[int32](1)
+				},
+				gomega.Succeed(),
+			),
+			ginkgo.Entry("can update maximum execution time when not admitted",
+				func() *kueue.Workload {
+					return testing.MakeWorkload(workloadName, ns.Name).
+						MaximumExecutionTimeSeconds(1).
+						Obj()
+				},
+				false,
+				func(newWL *kueue.Workload) {
+					*newWL.Spec.MaximumExecutionTimeSeconds = 2
+				},
+				gomega.Succeed(),
+			),
+			ginkgo.Entry("cannot add maximum execution time when not admitted",
+				func() *kueue.Workload {
+					return testing.MakeWorkload(workloadName, ns.Name).Obj()
+				},
+				true,
+				func(newWL *kueue.Workload) {
+					newWL.Spec.MaximumExecutionTimeSeconds = ptr.To[int32](1)
+				},
+				testing.BeAPIError(testing.InvalidError),
+			),
+			ginkgo.Entry("cannot update maximum execution time when not admitted",
```

Comment:
```suggestion
			ginkgo.Entry("cannot update maximum execution time when admitted",
```

It seems that this case has QuotaReservations.

Reply:
done, thanks